packages/jest-cli/src/TestNamePatternPrompt.js

```diff
@@ -81,12 +83,20 @@ module.exports = class TestNamePatternPrompt {
       this._pipe.write(' from cached test suites.');
 
       const width = getTerminalWidth();
+      const { start, end, index } = scroll(total, options);
 
-      results.forEach(name => {
-        const testName = formatTestNameByPattern(name, pattern, width - 4);
+      this._prompt.setTypeaheadLength(total);
 
-        this._pipe.write(`\n ${chalk.dim('\u203A')} ${testName}`);
-      });
+      matchedTests.slice(start, end)
+      .map(name => formatTestNameByPattern(name, pattern, width - 4))
+      .map((item, i) => {
+        if (i === index) {
+          this._prompt.setTypheadheadSelection(chalk.stripColor(item));
```

Comment:
Should this be `this._prompt.setTypheadheadSelection('^' + chalk.stripColor(item) + '$');` instead?